pkg/utils/container.go

```diff
@@ -290,14 +288,32 @@ func SaveImage(image string) (io.ReadCloser, error) {
 	return reader, err
 }
 
-// SaveImageAndExtract from docker to outputDir only for path defaultEvePrefixInTar in docker rootfs
-func SaveImageAndExtract(image, outputDir, defaultEvePrefixInTar string) error {
-	reader, err := SaveImage(image)
+// ExtractFromImage creates a container from an image, copies a file or directory from it, and then removes the container.
+func ExtractFromImage(imageName, localPath, containerPath string) error {
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
 	if err != nil {
-		return err
+		return fmt.Errorf("client.NewClientWithOpts: %w", err)
+	}
+
+	// Create a temporary container
+	resp, err := cli.ContainerCreate(ctx, &container.Config{
+		Image: imageName,
+	}, nil, nil, nil, "")
+	if err != nil {
+		return fmt.Errorf("error creating container: %w", err)
+	}
+	containerID := resp.ID
+	defer cli.ContainerRemove(ctx, containerID, types.ContainerRemoveOptions{Force: true})
```

Comment:
What if we can't remove container? Will it stay zombie?

Reply:
not really a zombie, since we never run the container and only created it, it will just stay inactive - which shouldn't really happen since there is nothing to stop (no process is running) and we force the removal to prevent minor issues from popping up.

the proper way would be of course to handle the error that the function might return, but we don't do that when closing files and probably shouldn't do here, since it will make the code much less readable